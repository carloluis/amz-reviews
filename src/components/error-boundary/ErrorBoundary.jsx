import React from 'react';
import PropTypes from 'prop-types';
import styles from './ErrorBoundary.scss';

class ErrorBoundary extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            error: '',
            info: ''
        };
    }
    componentDidCatch(error, info) {
        this.setState({
            error: error.message,
            info
        });
    }
    render() {
        if (this.state.error) {
            return (
                <div className={styles.container}>
                    <h2>Something went wrong!</h2>
                    <details>
                        {this.state.error}
                        <br />
                        {this.state.info.componentStack}
                    </details>
                </div>
            );
        }

        return this.props.children;
    }
}

ErrorBoundary.propTypes = {
    children: PropTypes.node.isRequired
};

export default ErrorBoundary;
