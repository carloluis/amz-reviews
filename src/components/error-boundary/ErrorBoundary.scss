@import '../../styles/palette.scss';

.container {
    border: 1px dashed $dark-font;
    padding: 8px;
    color: $dark-font;

    & > details {
        cursor: pointer;
    }
}
