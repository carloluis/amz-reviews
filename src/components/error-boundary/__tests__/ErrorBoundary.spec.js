import React from 'react';
import renderer from 'react-test-renderer';
import ErrorBoundary from '../ErrorBoundary';

const PROPS = {
    children: 'some text'
};

describe('<Search/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<ErrorBoundary {...PROPS} />).toJSON();
        expect(result).toMatchSnapshot();
    });
});
