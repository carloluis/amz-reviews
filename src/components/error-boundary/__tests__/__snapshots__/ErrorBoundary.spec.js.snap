// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> snapshot 1`] = `"some text"`;
